### ISSUE TITLE
Putting .scripts index takes `index.number_of_shards` into account

### ISSUE THREAD

**Issue by spinscale** [elastic/elasticsearch#18912]
**Elasticsearch version**: 2.3.2

**JVM version**: 1.8.0_60

**OS version**: osx

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:
 1. Set `index.number_of_shards: 3` in `config/elasticsearch.yml`
 2. Create the scripts index with a single shard
```
DELETE /.scripts/

PUT /.scripts/
{
  "settings": {
    "index": {
      "number_of_shards": 1,
      "number_of_replicas": 0
    }
  }
}
```
 3. Call `GET /.scripts/` and see there are three shards for this index.

This does not happen with index templates it seems.
Also regular indexes are created correctly with one shard.

**Comment by dadoonet** [comment]
> This does not happen with index templates it seems.

Using templates is the way to go IMO.
If I'm not mistaken, in 5.0, you can't set anymore index level settings in `elasticsearch.yml`.

**Comment by martijnvg** [comment]
The problem is here:
https://github.com/elastic/elasticsearch/blob/b2c4c323e1014563dd02e9cc2f3a9175fd831e78/core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java#L279

When creating `.scripts` index, it always overrides the number of shards settings regardless if it has been specified in the create index request.

**Comment by martijnvg** [comment]
Fixed via #18965
